dev/NavigationView/NavigationView.cpp

```diff
@@ -3086,17 +3086,78 @@ void NavigationView::UpdateTopNavListViewItemSource(const winrt::IInspectable& i
 
 void NavigationView::UpdateListViewItemSource()
 {
-    if (!m_appliedTemplate)
-    {
-        return;
-    }
-
     auto dataSource = MenuItemsSource();
     if (!dataSource)
     {
         dataSource = MenuItems();
     }
 
+    auto iterable = dataSource.try_as<winrt::IIterable<winrt::IInspectable>>();
+    if (iterable)
+    {
+        // Certain items are disallowed in a NavigationView's items list. Check for them.
+        auto checkItemIsValid = [](const winrt::IInspectable& item)
+        {
+            auto wuxItem = item.try_as<winrt::Windows::UI::Xaml::Controls::NavigationViewItem>();
+            if (wuxItem)
```

Comment:
>wuxItem [](start = 16, length = 7)

maybe just item.try_as<winrt::Windows::UI::Xaml::Controls::NavigationViewItemBase>()

Reply:
And remove the unused variable? Sure.